packages/devextreme-cli/src/templates-ts/react/application/src/contexts/auth.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect, createContext, useContext, useCallback } from 'react';
+import { getUser, signIn as sendSignInRequest } from '../api/auth';
+
+interface IUser {
+  email: string;
+  avatarUrl: string;
+}
+
+type IAuthContextType = {
+  user?: IUser;
+  signIn: (email: string, password: string) => Promise<{isOk: boolean, data?: IUser, message?: string}>;
+  signOut: () => void;
+  loading: boolean;
+}
+
+interface IAuthProviderProps {
+  children: React.ReactNode;
+}
+
+function AuthProvider(props: IAuthProviderProps) {
+  const [user, setUser] = useState<IUser>();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    (async function () {
+      const result = await getUser();
+      if (result.isOk) {
+        setUser(result.data);
+      }
+
+      setLoading(false);
+    })();
+  }, []);
+
+  const signIn = useCallback(async (email: string, password: string) => {
```

Comment:
If setUser doesn't mutate we should the useCallback if mutate we should add it to the dependancies

Reply:
It isn't my code, it is copy past. I will research why it was made like this in the next pr